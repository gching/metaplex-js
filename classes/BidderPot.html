<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>BidderPot | metaplex-js</title>
	<meta name="description" content="Documentation for metaplex-js">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
	<script async src="../assets/js/search.js" id="search-script"></script>
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">metaplex-js</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../modules.html">metaplex-js</a>
				</li>
				<li>
					<a href="BidderPot.html">BidderPot</a>
				</li>
			</ul>
			<h1>Class BidderPot</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel tsd-hierarchy">
				<h3>Hierarchy</h3>
				<ul class="tsd-hierarchy">
					<li>
						<span class="target">BidderPot</span>
					</li>
				</ul>
			</section>
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Constructors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-constructor tsd-parent-kind-class"><a href="BidderPot.html#constructor" class="tsd-kind-icon">constructor</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Properties</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="BidderPot.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="BidderPot.html#auctionAct" class="tsd-kind-icon">auction<wbr>Act</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="BidderPot.html#bidderAct" class="tsd-kind-icon">bidder<wbr>Act</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="BidderPot.html#bidderPot" class="tsd-kind-icon">bidder<wbr>Pot</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="BidderPot.html#emptied" class="tsd-kind-icon">emptied</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><a href="BidderPot.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Accessors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-get-signature tsd-parent-kind-class"><a href="BidderPot.html#isEmptied" class="tsd-kind-icon">is<wbr>Emptied</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Methods</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><a href="BidderPot.html#decode" class="tsd-kind-icon">decode</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Constructors</h2>
				<section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class">
					<a name="constructor" class="tsd-anchor"></a>
					<h3>constructor</h3>
					<ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon">new <wbr>Bidder<wbr>Pot<span class="tsd-signature-symbol">(</span>args<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="BidderPot.html" class="tsd-signature-type" data-tsd-kind="Class">BidderPot</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;isEmptied&quot;</span><span class="tsd-signature-symbol">&gt;</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="BidderPot.html" class="tsd-signature-type" data-tsd-kind="Class">BidderPot</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L32">models/programs/auction/BidderPot.ts:32</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>args: <span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="BidderPot.html" class="tsd-signature-type" data-tsd-kind="Class">BidderPot</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;isEmptied&quot;</span><span class="tsd-signature-symbol">&gt;</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="BidderPot.html" class="tsd-signature-type" data-tsd-kind="Class">BidderPot</a></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Properties</h2>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="accountPublicKey" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagOptional">Optional</span> account<wbr>Public<wbr>Key</h3>
					<div class="tsd-signature tsd-kind-icon">account<wbr>Public<wbr>Key<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L22">models/programs/auction/BidderPot.ts:22</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="auctionAct" class="tsd-anchor"></a>
					<h3>auction<wbr>Act</h3>
					<div class="tsd-signature tsd-kind-icon">auction<wbr>Act<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L26">models/programs/auction/BidderPot.ts:26</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="bidderAct" class="tsd-anchor"></a>
					<h3>bidder<wbr>Act</h3>
					<div class="tsd-signature tsd-kind-icon">bidder<wbr>Act<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L25">models/programs/auction/BidderPot.ts:25</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="bidderPot" class="tsd-anchor"></a>
					<h3>bidder<wbr>Pot</h3>
					<div class="tsd-signature tsd-kind-icon">bidder<wbr>Pot<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L24">models/programs/auction/BidderPot.ts:24</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="emptied" class="tsd-anchor"></a>
					<h3>emptied</h3>
					<div class="tsd-signature tsd-kind-icon">emptied<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L27">models/programs/auction/BidderPot.ts:27</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-static">
					<a name="SCHEMA" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> SCHEMA</h3>
					<div class="tsd-signature tsd-kind-icon">SCHEMA<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BorshSchema</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L7">models/programs/auction/BidderPot.ts:7</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Accessors</h2>
				<section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class">
					<a name="isEmptied" class="tsd-anchor"></a>
					<h3>is<wbr>Emptied</h3>
					<ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> isEmptied<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L29">models/programs/auction/BidderPot.ts:29</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">boolean</span></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Methods</h2>
				<section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class tsd-is-static">
					<a name="decode" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> decode</h3>
					<ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class tsd-is-static">
						<li class="tsd-signature tsd-kind-icon">decode<span class="tsd-signature-symbol">(</span>accountWithPubKey<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AccountWithPublicKey</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="BidderPot.html" class="tsd-signature-type" data-tsd-kind="Class">BidderPot</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/auction/BidderPot.ts#L36">models/programs/auction/BidderPot.ts:36</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>accountWithPubKey: <span class="tsd-signature-type">AccountWithPublicKey</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="BidderPot.html" class="tsd-signature-type" data-tsd-kind="Class">BidderPot</a></h4>
						</li>
					</ul>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class=" ">
						<a href="../modules.html">Exports</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
				<ul class="current">
					<li class="current tsd-kind-class">
						<a href="BidderPot.html" class="tsd-kind-icon">Bidder<wbr>Pot</a>
						<ul>
							<li class=" tsd-kind-constructor tsd-parent-kind-class">
								<a href="BidderPot.html#constructor" class="tsd-kind-icon">constructor</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="BidderPot.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="BidderPot.html#auctionAct" class="tsd-kind-icon">auction<wbr>Act</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="BidderPot.html#bidderAct" class="tsd-kind-icon">bidder<wbr>Act</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="BidderPot.html#bidderPot" class="tsd-kind-icon">bidder<wbr>Pot</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="BidderPot.html#emptied" class="tsd-kind-icon">emptied</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class tsd-is-static">
								<a href="BidderPot.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a>
							</li>
							<li class=" tsd-kind-get-signature tsd-parent-kind-class">
								<a href="BidderPot.html#isEmptied" class="tsd-kind-icon">is<wbr>Emptied</a>
							</li>
							<li class=" tsd-kind-method tsd-parent-kind-class tsd-is-static">
								<a href="BidderPot.html#decode" class="tsd-kind-icon">decode</a>
							</li>
						</ul>
					</li>
				</ul>
				<ul class="after-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>