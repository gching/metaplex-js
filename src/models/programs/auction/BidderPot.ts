import { PublicKey } from '@solana/web3.js';
import { deserializeUnchecked } from 'borsh';
import { BorshSchema } from '../../utils/types';
import { AccountWithPublicKey } from '../../../utils/privateTypes';

export class BidderPot {
  static SCHEMA: BorshSchema = new Map([
    [
      BidderPot,
      {
        kind: 'struct',
        fields: [
          ['bidderPot', 'pubkey'],
          ['bidderAct', 'pubkey'],
          ['auctionAct', 'pubkey'],
          ['emptied', 'u8'],
        ],
      },
    ],
  ]);

  accountPublicKey?: PublicKey;
  /// Points at actual pot that is a token account
  bidderPot: PublicKey;
  bidderAct: PublicKey;
  auctionAct: PublicKey;
  emptied: 0 | 1;

  get isEmptied(): boolean {
    return this.emptied === 1;
  }
  constructor(args: Omit<BidderPot, 'isEmptied'>) {
    Object.assign(this, args);
  }

  static decode(accountWithPubKey: AccountWithPublicKey): BidderPot {
    const bidderPot = deserializeUnchecked(
      BidderPot.SCHEMA,
      BidderPot,
      accountWithPubKey.account.data,
    ) as BidderPot;

    return new BidderPot({
      ...bidderPot,
      accountPublicKey: accountWithPubKey.pubkey,
    });
  }
}
