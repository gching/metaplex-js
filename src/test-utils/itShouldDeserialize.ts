import { PublicKey } from '@solana/web3.js';
import { serialize } from 'borsh';
import { BorshSchema } from '../models/utils/types';

export function itShouldDeserialize<E>(
  cls: {
    SCHEMA: BorshSchema;
    // eslint-disable-next-line @typescript-eslint/ban-types
    decode: Function;
  },
  received: E,
  accountPubKey = PublicKey.default,
): void {
  const data = Buffer.from(serialize(cls.SCHEMA, received));

  const deserialized = cls.decode({
    pubkey: accountPubKey,
    account: {
      data,
    },
  }) as E;

  expect(deserialized).toDeepEqual(received);
}
