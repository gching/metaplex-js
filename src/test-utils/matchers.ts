import { isEqualWith } from 'lodash';
import BN from 'bn.js';

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    interface Matchers<R> {
      toDeepEqual: <E = any>(expected: E) => R;
    }
  }
}

expect.extend({
  toDeepEqual<E = any>(received: E, expected: E) {
    const pass = isEqualWith(received, expected, (objValue, otherValue) => {
      if (BN.isBN(objValue) && BN.isBN(otherValue)) {
        return objValue.eq(otherValue);
      }
      return undefined;
    });

    if (pass) {
      return {
        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
        message: () => `expected ${received} not to deep equal ${expected}`,
        pass: true,
      };
    }

    return {
      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
      message: () => `expected ${received} to deep equal ${expected}`,
      pass: false,
    };
  },
});

export default undefined;
