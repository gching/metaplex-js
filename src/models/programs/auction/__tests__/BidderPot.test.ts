import { PublicKey } from '@solana/web3.js';
import { BidderPot } from '../BidderPot';
import { itShouldDeserialize } from '../../../../test-utils/itShouldDeserialize';

describe('BidderPot', () => {
  it('should instantiate with the correct values', () => {
    const bidderPot = new BidderPot({
      bidderPot: PublicKey.default,
      bidderAct: PublicKey.default,
      auctionAct: PublicKey.default,
      emptied: 0,
    });

    expect(bidderPot).toMatchSnapshot();
  });

  it('should deserialize with the correct values', () => {
    const bidderPot = new BidderPot({
      bidderPot: PublicKey.default,
      bidderAct: PublicKey.default,
      auctionAct: PublicKey.default,
      emptied: 0,
    });

    bidderPot.accountPublicKey = PublicKey.default;

    itShouldDeserialize(BidderPot, bidderPot);
  });
});
