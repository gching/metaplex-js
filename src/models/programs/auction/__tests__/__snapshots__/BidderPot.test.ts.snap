// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidderPot should instantiate with the correct values 1`] = `
BidderPot {
  "auctionAct": PublicKey {
    "_bn": "00",
  },
  "bidderAct": PublicKey {
    "_bn": "00",
  },
  "bidderPot": PublicKey {
    "_bn": "00",
  },
  "emptied": 0,
}
`;
